### PR TITLE
Create is_hvac_sys_fluid_loop_purchased_heating.md

### PR DESCRIPTION
**Pull request by claperle**

### CODE REVIEW

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ 
+**Function Call:** None  
+
+## Logic:   
+- Check if RMR is not modeled with external fluid source: `if NOT RMR.ASHRAE229.external_fluid_source: is_fluid_loop_purchased_heating = FALSE`
+- Else, continue on: `Else:`  
+    - for each external fluid source in RMR: `for external_fluid_source.id in RMR.ASHRAE229.external_fluid_source:`
+        - Check if external fluid source is heating type: `if external_fluid_source.id.type == "HOT_WATER"` QUESTION: SHOULD IT IT BE OR STEAM?  
+            - Add the associated loop to the purchased_hot_water_loop_list_b list: `purchased_hot_water_loop_list_b = purchased_hot_water_loop_list_b.append (external_fluid_source.id.type)` 
+    - Create an object associate with the heating_system fluid loop associated with hvac_b: `fluid_loop_b = hvac_b.heating_system[0].hot_water_loop`
```
Comment (supriyagoel) [inline docs/ruleset_functions/is_fluid_loop_purchased_heating.md:20]: same comment as https://github.com/pnnl/ruleset-checking-tool/pull/493
need a step to verify that heating system type = hot water loop

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+**Returns:**  
+- **is_fluid_loop_purchased_heating**: Returns TRUE if the fluid loop associated with the heating system associated with the HVAC system is attached to an external purchased heating loop. Returns FALSE if this is not the case.   
+ 
+**Function Call:** None  
+
+## Logic:   
+- Check if RMR is not modeled with external fluid source: `if NOT RMR.ASHRAE229.external_fluid_source: is_fluid_loop_purchased_heating = FALSE`
+- Else, continue on: `Else:`  
+    - for each external fluid source in RMR: `for external_fluid_source.id in RMR.ASHRAE229.external_fluid_source:`
+        - Check if external fluid source is heating type: `if external_fluid_source.id.type == "HOT_WATER"` QUESTION: SHOULD IT IT BE OR STEAM?
```
Comment (supriyagoel) [inline docs/ruleset_functions/is_fluid_loop_purchased_heating.md:18]: good point. @mkarpman please confirm if it should be HW or steam.

**Review by supriyagoel — COMMENTED** [review]
same comment as https://github.com/pnnl/ruleset-checking-tool/pull/493

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md (hunk 1 above)
Comment (claperle) [reply docs/ruleset_functions/is_fluid_loop_purchased_heating.md:20]: @supriyagoel This function won't be called until that is the case from the https://github.com/pnnl/ruleset-checking-tool/pull/480/files funciton.

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- **B-RMR**: To evaluate if the fluid loop associated with the heating system associated with the HVAC system is attached to an external purchased heating loop.   
+- **hvac_b.id**: The id of the hvac system to evaluate.  
+
+**Returns:**  
+- **is_fluid_loop_purchased_heating**: Returns TRUE if the fluid loop associated with the heating system associated with the HVAC system is attached to an external purchased heating loop. Returns FALSE if this is not the case.   
+ 
+**Function Call:** None  
+
+## Logic:   
+- Check if RMR is not modeled with external fluid source: `if NOT RMR.ASHRAE229.external_fluid_source: is_fluid_loop_purchased_heating = FALSE`
```
Comment (mkarpman) [inline docs/ruleset_functions/is_fluid_loop_purchased_heating.md:15]: Should be RulesetModelInstance.external_fluid_source. This applies to line 17 as well.

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md (hunk 2 above)
Comment (mkarpman) [reply docs/ruleset_functions/is_fluid_loop_purchased_heating.md:18]: Yes, it should be either HW or steam. 
G3.1.1.1 Purchased Heat
For systems using purchased hot water or steam, the heating source shall be modeled as purchased hot water or steam in both the proposed design and baseline building design.

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md (hunk 3 above)
Comment (claperle) [reply docs/ruleset_functions/is_fluid_loop_purchased_heating.md]: Corrected

File: docs/ruleset_functions/is_fluid_loop_purchased_heating.md (hunk 2 above)
Comment (claperle) [reply docs/ruleset_functions/is_fluid_loop_purchased_heating.md]: Updated

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- **hvac_b.id**: The id of the hvac system to evaluate.  
+
+**Returns:**  
+- **is_hvac_sys_fluid_loop_purchased_heating**: Returns TRUE if the fluid loop associated with the heating system associated with the HVAC system is attached to an external purchased heating loop. Returns FALSE if this is not the case.   
+ 
+**Function Call:** None  
+
+## Logic:   
+- Set is_hvac_sys_fluid_loop_purchased_heating equal to FALSE: `is_hvac_sys_fluid_loop_purchased_heating = FALSE`  
+- Check if RMR is not modeled with external fluid source: `if RMR.RuleModelInstance.external_fluid_source != Null:`
```
Comment (supriyagoel) [inline docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:16]: does 'not' need to be removed here? same comment as #495

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+**Function Call:** None  
+
+## Logic:   
+- Set is_hvac_sys_fluid_loop_purchased_heating equal to FALSE: `is_hvac_sys_fluid_loop_purchased_heating = FALSE`  
+- Check if RMR is not modeled with external fluid source: `if RMR.RuleModelInstance.external_fluid_source != Null:`  
+    - For each external fluid source in RMR: `for external_fluid_source.id in RMR.RuleModelInstance.external_fluid_source:`  
+        - Check if external fluid source is heating type: `if external_fluid_source.id.type == "HOT_WATER" OR external_fluid_source.id.type == "STEAM":`    
+            - Add the associated loop to the purchased_heating_loop_list_b list: `purchased_heating_loop_list_b = purchased_heating_loop_list_b.append (external_fluid_source.id)`   
+    - Create an object associate with the heating_system fluid loop associated with hvac_b: `fluid_loop_b = hvac_b.heating_system[0].hot_water_loop`
+    - Check if the fluid loop id is in the list created above, if yes then is_hvac_sys_fluid_loop_purchased_heating equals TRUE  : `if fluid_loop_b.id in purchased_heating_loop_list_b: is_hvac_sys_fluid_loop_purchased_heating = TRUE`
```
Comment (supriyagoel) [inline docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:21]: same questions as #495 - should we be expecting more than one loop or flagging for the same?

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md (hunk 4 above)
Comment (claperle) [reply docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:16]: Agreed. That is a relic. Correct.

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md (hunk 5 above)
Comment (claperle) [reply docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:21]: @supriyagoel I was thinking there could be more than one external loop if it is for some sort of process heating function.

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ 
+**Function Call:** None  
+
+## Logic:   
+- Set is_hvac_sys_fluid_loop_purchased_heating equal to FALSE: `is_hvac_sys_fluid_loop_purchased_heating = FALSE`  
+- Check if RMR is modeled with external fluid source: `if RMR.RuleModelInstance.external_fluid_source != Null:`  
+    - For each external fluid source in RMR: `for external_fluid_source.id in RMR.RuleModelInstance.external_fluid_source:`  
+        - Check if external fluid source is heating type: `if external_fluid_source.id.type == "HOT_WATER" OR external_fluid_source.id.type == "STEAM":`    
+            - Add the associated loop to the purchased_heating_loop_list_b list: `purchased_heating_loop_list_b = purchased_heating_loop_list_b.append (external_fluid_source.id)`   
+    - Create an object associate with the heating_system fluid loop associated with hvac_b: `fluid_loop_b = hvac_b.heating_system[0].hot_water_loop`
```
Comment (claperle) [inline docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:20]: Update if needed depending on the HW/CHW coil decision (system or terminal Q).

File: docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md (hunk 5 above)
Comment (mkarpman) [reply docs/ruleset_functions/is_hvac_sys_fluid_loop_purchased_heating.md:21]: This function is meant to determine whether a given HVAC system is served by purchased HW loop.  21-16 checks that baseline has only one HW plant, so should flag RMDs with multiple HW loops.

**Review by mkarpman — APPROVED** [review]

**Review by supriyagoel — APPROVED** [review]

**Review by supriyagoel — APPROVED** [review]

**Pull request merged**